src/Registration/cReg/cReg.cpp

```diff
@@ -447,13 +447,16 @@ void* cReg_NiftiImageData_get_inner_product(const void* im1_ptr, const void* im2
     CATCH;
 }
 extern "C"
-void* cReg_NiftiImageData_from_SIRFImageData(void* ptr)
+void* cReg_NiftiImageData_from_SIRFImageData(void* ptr, const int is_3D)
 {
-	try {
+    try {
         ImageData& sirf_im = objectFromHandle<ImageData>(ptr);
-        std::shared_ptr<NiftiImageData<float> >
-            sptr(new NiftiImageData<float>(sirf_im));
-        return newObjectHandle(sptr);
+
+        // Use an int as if bool to optionally run 3D checks
```

Comment:
cryptic!

Reply:
I maybe I understand, maybe we can't wrap `bool`? fine. I'd put this in the doxygen documentation  of the wrapper then

Reply:
Yeah, as far as I know we can pass `int` without any wrapping, but I don't think the same is true for booleans.